### PR TITLE
MIRROR - Fixes runtime from attempting checking armor on missing bodypart, improves ran_zone selection mechanism #43222

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
[Changelogs]:

https://github.com/tgstation/tgstation/pull/43222/files#

[why]
Says it fixes shit that will maybe help with lag

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
***I really suggest a closer look at this to confirm it***

**Comment by deathride58** [comment]
This won't help with lag at all. All it'll do is fix the bug where attacks sometimes don't work at all against people who are missing a limb.

File: code/modules/mob/mob_helpers.dm [hunk 1]
```diff
@@ -29,31 +29,12 @@
...
-		if(7 to 10)
-			return BODY_ZONE_R_ARM
-		if(11 to 14)
-			return BODY_ZONE_L_LEG
-		if(15 to 18)
-			return BODY_ZONE_R_LEG
 
-	return zone
-
-/proc/above_neck(zone)
```
Comment (deathride58) [inline code/modules/mob/mob_helpers.dm]: Why is this being removed? It wasn't touched at all in the original PR.

**Review by deathride58 — CHANGES_REQUESTED** [review]

File: code/modules/mob/mob_helpers.dm (hunk 1 above)
Comment (Trilbyspaceclone) [reply code/modules/mob/mob_helpers.dm]: HECK

**Review by deathride58 — APPROVED** [review]

**Pull request merged**